Patch-Source: https://sources.debian.org/data/main/q/qstardict/2.0.2-1/debian/patches/000-qt5-to-qt6.patch
The first five hunks were extracted to fix invalid conversion errors.
--
diff -Nru qstardict-2.0.2.orig/plugins/stardict/lib.cpp qstardict-2.0.2/plugins/stardict/lib.cpp
--- qstardict-2.0.2.orig/plugins/stardict/lib.cpp	2024-05-18 16:25:05.000000000 +0000
+++ qstardict-2.0.2/plugins/stardict/lib.cpp	2024-09-15 17:29:52.806038956 +0000
@@ -77,7 +77,7 @@
     }
 
     p3 = strchr(p2 + sizeof("\nwordcount=") - 1, '\n');
-    gchar *tmpstr = (gchar *)g_memdup(p2 + sizeof("\nwordcount=") - 1, p3 - (p2 + sizeof("\nwordcount=") - 1) + 1);
+    gchar *tmpstr = (gchar *)g_memdup2(p2 + sizeof("\nwordcount=") - 1, p3 - (p2 + sizeof("\nwordcount=") - 1) + 1);
     tmpstr[p3 - (p2 + sizeof("\nwordcount=") - 1)] = '\0';
     wordcount = atol(tmpstr);
     g_free(tmpstr);
@@ -91,7 +91,7 @@
             return false;
         }
         p3 = strchr(p2 + sizeof("\ntdxfilesize=") - 1, '\n');
-        tmpstr = (gchar *)g_memdup(p2 + sizeof("\ntdxfilesize=") - 1, p3 - (p2 + sizeof("\ntdxfilesize=") - 1) + 1);
+        tmpstr = (gchar *)g_memdup2(p2 + sizeof("\ntdxfilesize=") - 1, p3 - (p2 + sizeof("\ntdxfilesize=") - 1) + 1);
         tmpstr[p3 - (p2 + sizeof("\ntdxfilesize=") - 1)] = '\0';
         index_file_size = atol(tmpstr);
         g_free(tmpstr);
@@ -107,7 +107,7 @@
         }
 
         p3 = strchr(p2 + sizeof("\nidxfilesize=") - 1, '\n');
-        tmpstr = (gchar *)g_memdup(p2 + sizeof("\nidxfilesize=") - 1, p3 - (p2 + sizeof("\nidxfilesize=") - 1) + 1);
+        tmpstr = (gchar *)g_memdup2(p2 + sizeof("\nidxfilesize=") - 1, p3 - (p2 + sizeof("\nidxfilesize=") - 1) + 1);
         tmpstr[p3 - (p2 + sizeof("\nidxfilesize=") - 1)] = '\0';
         index_file_size = atol(tmpstr);
         g_free(tmpstr);
@@ -982,7 +982,7 @@
     int iIndexCount = 0;
 
     for (guint32 i = 0; i < narticles() && iIndexCount < iBuffLen - 1; i++)
-        if (g_pattern_match_string(pspec, get_key(i)))
+        if (g_pattern_spec_match_string(pspec, get_key(i)))
             aIndex[iIndexCount++] = i;
 
     aIndex[iIndexCount] = -1; // -1 is the end.
@@ -1251,7 +1251,7 @@
         // Upper the first character and lower others.
         if (!bFound)
         {
-            gchar *nextchar = g_utf8_next_char(sWord);
+            gchar *nextchar = (gchar*)g_utf8_next_char(sWord);
             gchar *firstchar = g_utf8_strup(sWord, nextchar - sWord);
             nextchar = g_utf8_strdown(nextchar, -1);
             casestr = g_strdup_printf("%s%s", firstchar, nextchar);
